sw/device/lib/dif/dif_aes.h

```diff
@@ -0,0 +1,589 @@
+ * AES registers must be securely cleared, by calling `dif_aes_end`.
+ *
+ * Please see the following documentation for further information:
+ * https://docs.opentitan.org/hw/ip/aes/doc/
+ * https://csrc.nist.gov/csrc/media/publications/fips/197/final/documents/fips-197.pdf
+ * https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38a.pdf
+ */
+
+/**
+ * A typed representation of the AES key share.
+ *
+ * Two part masked AES key, where XOR operation of these two parts results in
+ * the actual key.
+ */
+typedef struct dif_aes_key_share {
+  /**
+   * One share of the key that when XORed with `part1` results in the actual
+   * key.
+   */
+  uint32_t part0[8];
+  /**
+   * One share of the key that when XORed with `part0` results in the actual
+   * key.
+   */
+  uint32_t part1[8];
+} dif_aes_key_share_t;
+
+/**
+ * A typed representation of the AES Initialisation Vector (IV).
+ */
+typedef struct dif_aes_iv {
+  uint32_t iv[4];
+} dif_aes_iv_t;
+
+/**
+ * A typed representation of the AES data.
+ */
+typedef struct dif_aes_data {
+  uint32_t data[4];
+} dif_aes_data_t;
+
+/**
+ * AES key length in bits.
+ */
+typedef enum dif_aes_key_length {
+  /**
+   * 128 bit wide AES key.
+   */
+  kDifAesKey128 = 0,
+  /**
+   * 192 bit wide AES key.
+   */
+  kDifAesKey192,
+  /**
+   * 256 bit wide AES key.
+   */
+  kDifAesKey256,
+} dif_aes_key_length_t;
+
+/**
+ * AES mode.
+ */
+typedef enum dif_aes_mode {
+  /**
+   * AES encryption mode.
+   */
+  kDifAesModeEncrypt = 0,
+  /**
+   * AES decryption mode.
+   */
+  kDifAesModeDecrypt,
+} dif_aes_mode_t;
+
+/**
+ * AES operation.
+ */
+typedef enum dif_aes_operation {
+  /**
+   * AES operates in automatic mode - which means that the encryption/decryption
+   * is automatically triggered on every successful `dif_aes_*_load_data()`.
+   */
+  kDifAesOperationAuto = 0,
+  /**
+   * AES operates in manual mode - which means that the encryption/decryption
+   * is manually triggered by `dif_aes_trigger(kDifAesTriggerStart)`.
+   */
+  kDifAesOperationManual,
+} dif_aes_operation_t;
+
+/**
+ * AES masking.
+ *
+ * NOTE:
+ * This should only be used for development purpose (SCA), and expected to be
+ * removed before the production version.
+ */
+typedef enum dif_aes_masking {
+  /**
+   * Pseudo-random generator is used for masking.
+   */
+  kDifAesMaskingInternalPrng = 0,
+  /**
+   * Completely disables masking by forcing all masks to zero.
+   */
+  kDifAesMaskingForceZero,
+} dif_aes_masking_t;
+
+/**
+ * Parameters for an AES transaction.
+ */
+typedef struct dif_aes_transaction {
+  dif_aes_key_length_t key_len;
+  dif_aes_mode_t mode;
+  dif_aes_operation_t operation;
+  dif_aes_masking_t masking;
+} dif_aes_transaction_t;
+
+/**
+ * An AES alert type.
+ */
+typedef enum dif_aes_alert {
+  /**
+   * Fatal alert conditions include i) storage errors in the Control Register,
+   * and ii) if any internal FSM enters an invalid state.
+   */
+  kDifAesAlertFatal = 0,
```

Comment:
Really sorry to bother you again about this. I had to rename it again in #4909 . It's now `kDifAlertFatalFault` and `kDifAlertRecovCtrlUpdateErr`.

Reply:
Done.